Lia reimplementing

LiaSolo opened the pull request:
- Add choose primitive page
- Add choose file input & modal

LiaSolo committed:
feat: port wizard layout, collapse, tooltip, const colors
fix: rename portal container, variables.module
create: choose file modal and input for it

LiaSolo committed:
feat: add choose primitive page

Pechenux committed:
fix: change header background style to remove collision with page background (#5)

LiaSolo committed:
feat: add store fo choose primitive and choose file pages
feat: add primitives enum
feat: create temp dataset upload page

LiaSolo committed:
feat: add store fo choose primitive and choose file pages
feat: add primitives enum
feat: create temp dataset upload page

LiaSolo committed:
fix: remove temp page

Pechenux commented on src/components/choose-file/ChooseFileModal/ChooseFileModal.tsx:
```diff
@@ -0,0 +1,170 @@
+import styles from './ChooseFileModal.module.scss';
+
+const ChooseFileModal: FC<ModalProps> = ({ isOpen, onClose }) => {
+  const builtinDatasets = [
```
Надо вынести в константы

Pechenux commented on src/components/choose-file/ChooseFileModal/ChooseFileModal.tsx:
```diff
@@ -0,0 +1,170 @@
+    },
+  ];
+
+  const userDatasets = [
```
Надо вынести в константы

Pechenux commented on src/components/choose-file/ChooseFileModal/ChooseFileModal.tsx:
```diff
@@ -0,0 +1,170 @@
+    },
+  ];
+
+  const header = (
```
Обернуть в useMemo такие компоненты, чтобы не генерировать их на каждом рендере

Pechenux commented on src/app/create-task/choose-primitive/choosePrimitive.tsx:
```diff
@@ -1,72 +1,93 @@
 
-  const handleClick = useCallback(async () => {
-    console.log(file);
+const header = (
```
Обернуть в useMemo header и footer

Pechenux commented on src/app/create-task/configure-algorithm/configureAlgorithm.tsx:
```diff
@@ -0,0 +1,56 @@
+    setIsOpen(true);
+  };
+
+  const header = (
```
Обернуть в usememo header и footer

Pechenux commented on src/app/create-task/configure-algorithm/configureAlgorithm.tsx:
```diff
@@ -0,0 +1,56 @@
+const ChoosePrimitive = () => {
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+
+  const onClose = () => {
```
useCallback onClose и handleFileClick

Pechenux replied:
handleClose так как это функция-обработчик

Pechenux commented on src/components/choose-primitive/CollapsableView/CollapsableView.module.scss:
```diff
@@ -0,0 +1,41 @@
+
+.collabsableOutput {
+  @include paragraph-small;
+  // overflow: hidden;
```
Кажется, лишнее

Pechenux commented on src/components/choose-primitive/PrimitiveCard/PrimitiveCard.tsx:
```diff
@@ -0,0 +1,68 @@
+    useAtom<PrimitiveType>(choosenPrimitiveAtom);
+
+  const isSelected = primitive === choosenPrimitive;
+  const { label, description, tags } = primitiveInfo[primitive] || {
```
Мне не нравится, что мы распаковываем данные на уровне комопнента. Давай это делать в компоненте страницы и там же будем проверять, выбран ли компонент.

Pechenux commented on src/components/choose-primitive/Tag/index.ts:
Пустой файл?

Pechenux commented on src/components/choose-primitive/PrimitiveCard/index.ts:
Пустой файл?

Pechenux commented on src/components/common/layout/Collapse/Collapse.module.scss:
```diff
@@ -0,0 +1,7 @@
+.title.collapsed img {
```
Давай не будем использовать теги компонентов в css так как это может поломать стили в других местах

Pechenux commented on src/components/common/layout/Collapse/Collapse.module.scss:
```diff
@@ -0,0 +1,7 @@
+  transform: rotate(180deg);
+}
+
+svg.reversed {
```
Для чего эти стили?

Pechenux commented on src/components/common/layout/Collapse/Collapse.tsx:
```diff
@@ -0,0 +1,27 @@
+import { WithChildren } from '@/types/withChildren';
+import styles from './Collapse.module.scss';
+
+type Props = WithChildren & {
```
Лучше CollapseProps

Pechenux commented on src/components/common/layout/Collapse/index.tsx:
```diff
@@ -0,0 +1,3 @@
+import { Collapse } from './Collapse';
```
export * from './Collapse';

Pechenux commented on src/components/common/layout/Header/Header.tsx:
```diff
@@ -1,29 +1,31 @@
 
 const HeaderComponent = () => {
-  const [headerBackground, setHeaderBackground] = useState(false);
+  // const [headerBackground, setHeaderBackground] = useState(false);
```
Это было решение для проблемы с хедером? Если так, то убери пожалуйста

Pechenux commented on src/components/common/layout/ModalContainer/ModalContainer.module.scss:
```diff
@@ -12,8 +12,8 @@
-  max-width: 560px;
-  width: 100%;
+  width: 560px;
+  //width: 100%;
```
?

Pechenux commented on src/components/common/layout/PropertiesModal/PropertiesModal.tsx:
```diff
@@ -0,0 +1,37 @@
+  );
+};
+
+export default PropertiesModal;
```
export const PropertiesModal

Pechenux commented on src/components/common/layout/PropertiesModal/index.ts:
```diff
@@ -0,0 +1 @@
+export { default } from './PropertiesModal';
```
export * from './PropertiesModal';

Pechenux commented on src/components/common/uikit/Inputs/File/File.tsx:
```diff
@@ -0,0 +1,50 @@
+    onClick?: () => void;
+  };
+
+const File: ForwardRefRenderFunction<HTMLInputElement, Props> = (
```
const FileComponent =

Pechenux commented on src/components/common/uikit/Inputs/File/File.tsx:
```diff
@@ -0,0 +1,50 @@
+  );
+};
+
+export default forwardRef(File);
```
export const File = forwardRef(FileComponent);

Pechenux commented on src/components/common/uikit/Inputs/File/index.ts:
```diff
@@ -0,0 +1 @@
+export { default } from './File';
```
export * from './File';

Pechenux commented on src/components/common/uikit/Inputs/MultiSelect/ControlledMultiSelect.tsx:
```diff
@@ -0,0 +1,50 @@
+  />
+);
+
+export default ControlledMultiSelect;
```
export const ControlledMultiSelect

Pechenux commented on src/components/common/uikit/Inputs/MultiSelect/MultiSelect.tsx:
```diff
@@ -0,0 +1,71 @@
+// eslint-disable-next-line  @typescript-eslint/no-explicit-any
+type RefElement = any;
+
+const MultiSelect: ForwardRefRenderFunction<RefElement, MultiSelectProps> = (
```
const MultiSelectComponent

Pechenux commented on src/components/common/uikit/Inputs/MultiSelect/MultiSelect.tsx:
```diff
@@ -0,0 +1,71 @@
+  );
+};
+
+export default forwardRef(MultiSelect) as <TValue = string>(
```
export const MultiSelect = forwardRef(MultiSelectComponent)

Pechenux commented on src/components/common/uikit/Inputs/MultiSelect/index.ts:
```diff
@@ -0,0 +1,2 @@
+export { default } from './MultiSelect';
```
export * в обоих случаях

Pechenux commented on src/components/common/uikit/Inputs/Search/Search.tsx:
```diff
@@ -0,0 +1,50 @@
+  );
+};
+
+export default forwardRef(Search);
```
export const = forwardRef

Pechenux commented on src/components/common/uikit/Inputs/Search/index.ts:
```diff
@@ -0,0 +1 @@
+export { default } from './Search';
```
export *

Pechenux commented on src/components/common/uikit/Tooltip/Tooltip.tsx:
```diff
@@ -0,0 +1,73 @@
+  );
+};
+
+export default Tooltip;
```
export const =

Pechenux commented on src/components/common/uikit/Tooltip/index.ts:
```diff
@@ -0,0 +1 @@
+export { default } from './Tooltip';
```
export *

LiaSolo committed:
feat: reinstall pnpm-lock.yaml

LiaSolo committed:
fix: default --> const
fix: file component --> select dataset component
fix: remove unused CollapsableView
feat: add store for select dataset component

LiaSolo committed:
fix: default --> const
fix: file component --> select dataset component
fix: remove unused CollapsableView
feat: add store for select dataset component
